ucf-email-editor-plugin.php

```diff
@@ -1,15 +1,36 @@
 <?php
 /*
 Plugin Name: UCF Email Editor Plugin
-Description: WordPress Plugin to add an email editor for creating email html content.
+Description: Provides email custom post type, email editor templates and related meta fields.
 Version: 1.0.0
 Author: UCF Web Communications
 License: GPL3
 GitHub Plugin URI: UCF/UCF-Email-Editor-Plugin
 */
 
 if ( ! defined( 'WPINC' ) ) {
-    die;
+	die;
 }
 
-include_once 'common/functions.php';
+define( 'UCF_EMAIL_EDITOR__PLUGIN_URL', plugins_url( basename( dirname( __FILE__ ) ) ) );
+define( 'UCF_EMAIL_EDITOR__PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
+define( 'UCF_EMAIL_EDITOR__PLUGIN_FILE', __FILE__ );
+
+include_once 'includes/ucf-email-posttype.php';
+include_once 'templates/templates.php';
+
+if ( ! function_exists( 'ucf_email_editor_plugin_activation' ) ) {
+	function ucf_email_editor_plugin_activation() {
```

Comment:
Adding a `flush_rewrite_rules()` call to plugin activation and deactivation hooks should help prevent issues with malfunctioning permalinks for Emails after activating/deactivating the plugin.

Examples: https://github.com/UCF/UCF-Degree-CPT-Plugin/blob/master/ucf-degree-cpt.php#L48-L67